src/utils/index.js

```diff
@@ -10,7 +14,35 @@ const hasAllKeys = (testObject, keys) => {
 
 const formatLogLevel = logLevel => ['debug', 'info', 'warn', 'error'][logLevel];
 
+const flattenObject = (source, prefix = '', refSet = new Set()) => {
+  // if type or source is primitive, return value directly
+  if (!isObject(source)) {
+    const result = {};
+    result[prefix || 'value'] = source;
+    return result;
+  }
+
+  // check for circular reference
+  if (refSet.has(source)) {
+    return {};
```

Comment:
How about returning a special string or something instead of `{}` to tell it encounters a circular reference